MH4U Dex/CoreDataController.m

```diff
@@ -363,6 +439,87 @@ - (NSUInteger)countForEntityWithEntityName:(NSString *)entityName withPredicate:
     }
 }
 
+- (QuestDrop *)questDropFromDictionary:(NSDictionary *)dropDict
+{
+    QuestDrop *drop = [QuestDrop insertInManagedObjectContext:self.managedObjectContext];
+    NSPredicate *questPredicate = [NSPredicate predicateWithFormat:@"%K == %@", QuestAttributes.id, dropDict[QuestDropQuestIDKey]];
+    drop.quest = (Quest *)[self getUniqueEntityWithEntityName:[Quest entityName] withPredicate:questPredicate];
+    if (!drop.quest) {
+        //If the quest was not found, then it makes no sense to store this reward.  Just return nil.
+        NSLog(@"The quest for a quest reward was not found!");
+        return nil;
+    }
```

Comment:
You will still have created the `QuestDrop` object `drop`, though—you might want to try to retrieve the quest first, check that it exists, then create `drop`.